filters/include/pcl/filters/impl/spherical_voxel_grid.hpp

```diff
@@ -0,0 +1,318 @@
+  bool operator < (const cloud_point_index_idx &p) const { return (idx < p.idx); }
+};
+
+template <typename PointT> void
+pcl::SphericalVoxelGrid<PointT>::applyFilter (PointCloud &output)
+{
+  // Has the input dataset been set already?
+  if (!input_)
+  {
+    PCL_WARN ("[pcl::%s::applyFilter] No input dataset given!\n", getClassName().c_str());
+    output.width = output.height = 0;
+    output.points.clear();
+    return;
+  }
+
+  // Copy the header (and thus the frame_id) + allocate enough space for points
+  output.height = 1;                    // downsampling breaks the organized structure
+  output.is_dense = true;                 // we filter out invalid points
+
+  if (leaf_size_r_ <= 0 || leaf_size_theta_ <= 0 || leaf_size_phi_ <= 0)
+  {
+    PCL_WARN("[pcl::%s::applyFilter] Leaf size improperly set.", getClassName().c_str());
+    output = *input_;
+    return;
+  }
+
+  // Find the farthest away point, subject to filter parameters if specified
+  Eigen::Vector4f max_p;
+
+  if (!filter_field_name_.empty ()) // If we don't want to process the entire cloud...
+  {
+    std::vector<int> indices_filtered;
+    indices_filtered.reserve(indices_->size());
+
+    // Get the distance field index
+    std::vector<pcl::PCLPointField> fields;
+    int filter_idx = pcl::getFieldIndex (*input_, filter_field_name_, fields);
+    if (filter_idx == -1)
+      PCL_WARN ("[pcl::%s::applyFilter] Invalid filter field name. Index is %d.\n", getClassName ().c_str (), filter_idx);
+
+    for (std::vector<int>::const_iterator it = indices_->begin (); it != indices_->end (); ++it)
+    {
+      if (!input_->is_dense)
+        // Check if the point is invalid
+        if (!pcl_isfinite (input_->points[*it].x) ||
+            !pcl_isfinite (input_->points[*it].y) ||
+            !pcl_isfinite (input_->points[*it].z))
+          continue;
+
+      // Get the filter value
+      const uint8_t* pt_data = reinterpret_cast<const uint8_t*> (&input_->points[*it]);
+      float filter_value = 0;
+      memcpy (&filter_value, pt_data + fields[filter_idx].offset, sizeof (float));
+
+      if (filter_limit_negative_)
+      {
+        // Use a threshold for cutting out points which inside the interval
+        if ((filter_value < filter_limit_max_) && (filter_value > filter_limit_min_))
+          continue;
+      }
+      else
+      {
+        // Use a threshold for cutting out points which are too close/far away
+        if ((filter_value > filter_limit_max_) || (filter_value < filter_limit_min_))
+          continue;
+      }
+
+      indices_filtered.push_back (*it);
+    }
+
+    getMaxDistance<PointT> (*input_, indices_filtered, filter_origin_, max_p);
+  }
+  else
+    getMaxDistance<PointT> (*input_, filter_origin_, max_p);
+
+  // Find the number of radial layers required given the farthest point and resolution
+  max_radius_ = (max_p - filter_origin_).norm ();
+  uint64_t rIdxNum = static_cast<uint64_t>(std::floor(max_radius_ / leaf_size_r_)) + 1;
```

Comment:
Why did you choose 64 bit here? And wouldn't it be more consistent to cast the division operands to `double` then?

Reply:
The purpose of this calculation is to verify that the number of discrete voxels does not overflow an `int32_t`. The multiplication within the if statement below is performed with `uint64_t`s to ensure that the potential overflow is detected. In this case, I just performed the cast here instead of within the if statement. I will move the cast into the if statement and switch `rIdxNum` to `int32_t` to make things more consistent.